src/aap_eda/tasks/analytics.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from datetime import datetime, timezone
+
+import django_rq
+from rq.exceptions import NoSuchJobError
+
+from aap_eda.analytics import collector
+from aap_eda.conf import application_settings
+from aap_eda.core.tasking import Job, unique_enqueue
+
+logger = logging.getLogger(__name__)
+
+
+ANALYTICS_SCHEDULE_JOB_ID = "gather_analytics"
+ANALYTICS_JOB_ID = "job_gather_analytics"
+ANALYTICS_TASKS_QUEUE = "default"
+
+
+def schedule_gather_analytics() -> None:
+    scheduler = django_rq.get_scheduler()
+    func = "aap_eda.tasks.analytics.gather_analytics"
+    scheduled_time = datetime.now(timezone.utc)
+    interval = application_settings.AUTOMATION_ANALYTICS_GATHER_INTERVAL
+    logger.info(
+        f"Adding periodic job {ANALYTICS_SCHEDULE_JOB_ID} "
+        f"to run every {interval} seconds"
+    )
+    scheduler.schedule(
+        scheduled_time=scheduled_time,
+        func=func,
+        interval=interval,
+        id=ANALYTICS_SCHEDULE_JOB_ID,
+    )
+
+
+def reschedule_gather_analytics(new_interval: int, serializer=None) -> None:
+    try:
+        job = Job.fetch(ANALYTICS_SCHEDULE_JOB_ID, serializer=serializer)
+    except NoSuchJobError:
+        logger.warning(f"Job {ANALYTICS_SCHEDULE_JOB_ID} does not exist")
```

Comment:
Why are you just skipping the action? I think we should call `schedule_gather_analytics`if the job doesn't exist, otherwise we lose the track of the change triggered by the system registry until next restart of the scheduler

Reply:
This is a normal case, for example, in a local dev env where only api is running, and application_settings.AUTOMATION_ANALYTICS_GATHER_INTERVAL is set to a new value.